ExampleMod/Items/Weapons/ExampleMagicMissile.cs

```diff
@@ -31,16 +32,30 @@ public class ExampleMagicMissile : ModItem
 		}
 
 		// This item's mana usage changes through the day, peaking at 1.5x mana usage at noon, and 0.5x mana usage at midnight.
-		public override void GetManaCost(Player player, ref int mana) {
+		public override void ModifyManaCost(Player player, ref float reduce, ref float mult) {
 			float currentTime = (float)Main.time;
+			// The time at which it changes from day to night and vice versa.
 			int maxTime = Main.dayTime ? 54000 : 32400;
+			// The time at which it is 12pm or 12am, which isn't exactly half the values above.
 			int time12 = Main.dayTime ? 28800 : 18000;
+			// If the time it after 12, recalculate the current time to account for the length of the second half of the day instead of the first.
 			if (currentTime > time12) {
+				currentTime -= time12;
 				time12 = maxTime - time12;
-				currentTime = maxTime - currentTime;
+				currentTime += time12;
 			}
-			float multi = 1 + currentTime / time12 * 0.5f * Main.dayTime.ToDirectionInt();
-			mana = (int)(mana * multi);
+			// If it is night time, add twice the duration of the current half of the night.
+			if (!Main.dayTime) {
+				currentTime += time12 * 2;
+			}
+			// We divide the current time by the current half of the day or night's time, then by 2.
+			// The result would gradually go from 0 at 4:30am, 0.5 at 12pm, 1 at 7:30pm and 1.5f at 12 am.
```

Comment:
you are correct here, but it's probably better to just have it be 1 at 6am and 6pm, the scaling to length of day/night is pretty confusing